### PR TITLE
add scala-java8-compat and jackson-module-scala to community build

### PR DESCRIPTION
**Pull request by pjfanning**
I tried to follow the steps in https://github.com/lampepfl/dotty/tree/master/community-build - feel free to request changes

### CODE REVIEW

**Review by griggt — COMMENTED** [review]
It seems that this has a binary dependency on the Scala 3 artifact `scala-java8-compat_3-1.0.0.jar`

The projects in the community build normally build all their Scala 3 dependencies from source, which would mean adding `scala-java8-compat` to the community build together here with `jackson-module-scala`.

wdyt @anatoliykmetyuk

**Comment by anatoliykmetyuk** [comment]
Yes, it would be good not to break the existing convention and also include the `scala-java8-compat` in this PR.

**Comment by pjfanning** [comment]
@anatoliykmetyuk @griggt I've added scala-java8-compat to this PR

File: community-build/src/scala/dotty/communitybuild/projects.scala [hunk 1]
```diff
@@ -792,6 +805,7 @@ def allProjects = List(
   projects.scalacheckEffect,
   projects.fs2,
   projects.libretto,
+  projects.jacksonModuleScala,
```
Comment (anatoliykmetyuk) [inline community-build/src/scala/dotty/communitybuild/projects.scala:808]: Why isn't the java8-compat included in this list?

Comment (pjfanning) [reply community-build/src/scala/dotty/communitybuild/projects.scala:809]: apologies - missed that, I've fixed up the PR to include that

**Comment by pjfanning** [comment]
the CI build is failing due to org.scala-lang:scala-compiler:3.0.3-RC1-bin-SNAPSHOT and other related jars not being found

**Comment by griggt** [comment]
> the CI build is failing due to org.scala-lang:scala-compiler:3.0.3-RC1-bin-SNAPSHOT and other related jars not being found

I've pushed a workaround

**Review by anatoliykmetyuk — APPROVED** [review]

**Pull request merged**